examples/Transfer_Learning/basic_transfer_learning.py

```diff
@@ -0,0 +1,175 @@
+## Transfer Learning
+
+# This example demonstrates BayBE's
+# {doc}`Transfer Learning </userguide/transfer_learning>` capabilities:
+# * We construct a campaign,
+# * give it access to data from a related but different task,
+# * and show how this additional information boosts optimization performance.
+
+### Imports
+
+import os
+import sys
+from pathlib import Path
+from typing import Dict, List
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from botorch.test_functions.synthetic import Hartmann
+
+from baybe import Campaign
+from baybe.objective import Objective
+from baybe.parameters import NumericalDiscreteParameter, TaskParameter
+from baybe.searchspace import SearchSpace
+from baybe.simulation import simulate_scenarios
+from baybe.targets import NumericalTarget
+from baybe.utils.botorch_wrapper import botorch_function_wrapper
+from baybe.utils.plotting import create_example_plots
+
+### Settings
+
+# The following settings are used to set up the problem:
+
+SMOKE_TEST = "SMOKE_TEST" in os.environ
+DIMENSION = 3  # input dimensionality of the test function
+BATCH_SIZE = 1  # batch size of recommendations per DOE iteration
+N_MC_ITERATIONS = 5 if SMOKE_TEST else 75  # number of Monte Carlo runs
+N_DOE_ITERATIONS = 5 if SMOKE_TEST else 10  # number of DOE iterations
+POINTS_PER_DIM = 5 if SMOKE_TEST else 5  # number of grid points per input dimension
+
+### Defining the Tasks
+
+# For our example, we consider the "Hartmann" test function.
+#
+# More specifically, to demonstrate the transfer learning mechanism, we consider the
+# problem of optimizing the function using training data from its negated version.
+# The used model is of course not aware of this relationship but needs to infer it
+# from the data gathered during the optimization process.
+
+test_functions = {
+    "Test_Function": botorch_function_wrapper(Hartmann(dim=DIMENSION)),
+    "Training_Function": botorch_function_wrapper(Hartmann(dim=DIMENSION, negate=True)),
```

Comment:
@AdrianSosic I still think it would b even nicer if this had also small amount of noise on top :) I understand that this is then "mixing multiple influences" but so what?

This is not an investigation, but a demonstration -> making it more real world is good

if you disagree this shouldnt be a blocker please just move on

Reply:
So just to be sure: The Training Function should have some noise, the Test Function should stay as it is? In that case, we could add small noise of maybe 5% of the total range of the Hartmann function (which is roughly 3.5, so a noise of around 0.15 might be reasonable).
All of this of course only if the results are convincing :D

Reply:
yes to all